README.rst

```diff
@@ -167,6 +167,50 @@ Without custom setting, users will be identified by the ``USERNAME_FIELD`` prope
 You can customize which field is used for the identifier by adding ``SAML_IDP_DJANGO_USERNAME_FIELD`` to your settings with as value the attribute to use on your user instance.
 You can also override this per SP by setting ``nameid_field`` in the SP config, as seen in the sample ``SAML_IDP_SPCONFIG`` above.
 
+User Agreement Screen
+=====================
+
+The User Agreement Screen shows users information about what is being sent to the SP, as well as information about the given SP. This can be toggled globally through the `SAML_IDP_SHOW_USER_AGREEMENT_SCREEN = True` setting. That can be overriden per-SP with `show_user_agreement_screen`.
+
+By default, the only information available about the SP is the entity id, which is often non-descriptive. You can manually add in a title and description through `display_name` and `display_description` respectively.
+
+By default, the user agreement screen will show all transmitted attributes to the user. If there are any attributes you wish to exclude, you can do so with the `SAML_IDP_USER_AGREEMENT_ATTR_EXCLUDE = []` setting. You can supplement this on a per-SP level with `user_agreement_attr_exclude`.
+
+An example setup can be found below::
+
+    SAML_IDP_SPCONFIG = {
+        'sample_sp_entity_id': {
+            'show_user_agreement_screen': False,
+        },
+        'sample_sp_2': { # User Agreement Screen will be shown because default is enabled below
+            'attribute_mapping': {
+                # DJANGO: SAML
+                'email': 'email',
+                'first_name': 'first_name',
+                'last_name': 'last_name',
+                'is_staff': 'is_staff',
+                'is_superuser':  'is_superuser',
+                'method_to_get_id': 'id',
+                'top_secret_attr': 'secret_attr', # This will not be shown to user as per default
+            },
+        },
+        'sample_sp_3': { # User Agreement Screen will be shown because default is enabled below
+            'attribute_mapping': {
+                # DJANGO: SAML
+                'sp_specific_secret_attr': 'sp_specific_secret_attr', # This will not be shown because specified in this sp conf
+                'top_secret_attr': 'secret_attr', # This will still not be shown because sp-specific excludes supplements, not overrides
+            },
+            'user_agreement_attr_exclude': ['sp_specific_secret_attr'],
```

Comment:
I should not insert custom arguments in the pySAML2 conf dictionary, I want to leave this as it come from pysaml2. 
We cannot exclude attributes, this is for GDPR compliance, is there some important goals you are reaching to exclude them? Tell me the use case please. Thank you, whatever done you made a good job

Reply:
> I should not insert custom arguments in the pySAML2 conf dictionary, I want to leave this as it come from pysaml2.

This is actually not the case. While `SAML_IDP_CONFIG` is indeed a pySAML2 conf dict, `SAML_IDP_SP_CONFIG` (which has been modified in this PR), is defined and used exclusively within djangosaml2idp's original code. As such, these modifications have nothing to do with pysaml2 config. 

> We cannot exclude attributes, this is for GDPR compliance, is there some important goals you are reaching to exclude them? Tell me the use case please.

Here's my rational:
1. GDPR is only applicable in EU countries.
2. As I've mentioned in previous threads, SAML (unlike OAuth) is generally used for internal IT networks and systems. Users will generally be employees, not the general public, and for employees, data processing reasons such as "legitimate interest of the employer" or "necessity for the performance of a contract" will more likely occur.
3. GDPR is primarily relevant to Personally Identifiable Information (PII). If a large quantity of attributes is being transmitted, being able to omit non-PII attributes from the user agreement screen would lead to a more straightforward, less confusing experience for users.
4. Using exclusion is completely optional, and disabled by default. I see no harm in having this potentially useful feature.